shuffle/cli/src/transactions.rs

```diff
@@ -0,0 +1,236 @@
+// SPDX-License-Identifier: Apache-2.0
+use crate::shared::{get_home_path, Home};
+use anyhow::{anyhow, Result};
+use diem_types::account_address::AccountAddress;
+use reqwest::{Client, Response};
+use serde_json::Value;
+use std::{cmp::max, fs, io, io::Write, str::FromStr, thread, time};
+use url::Url;
+
+// Will list the last 10 transactions and has the ability to block and stream future transactions.
+pub async fn handle(network: Url, tailing: bool) -> Result<()> {
+    let home = Home::new(get_home_path().as_path())?;
+    let address_str = fs::read_to_string(home.get_latest_address_path())?;
+    let address = AccountAddress::from_str(address_str.as_str())?;
+    let client = reqwest::Client::new();
+
+    let account_seq_num = get_account_sequence_number(&client, &network, address).await?;
+    let start = max(account_seq_num as i64 - 10, 0);
+
+    let resp = get_account_transactions_response(&client, address, &network, start, 10).await?;
+    let json_with_txns: serde_json::Value = serde_json::from_str(resp.text().await?.as_str())?;
+
+    let all_transactions = json_with_txns
+        .as_array()
+        .ok_or_else(|| anyhow::anyhow!("Failed to get transactions"))?;
+
+    for txn in all_transactions.iter() {
+        write_into(&mut io::stdout(), txn)?;
+        println!();
+        println!();
+    }
+
+    let mut old_seq_num = -1;
+    if !all_transactions.is_empty() {
+        old_seq_num = parse_txn_for_seq_num(
+            all_transactions
+                .last()
+                .ok_or_else(|| anyhow!("Couldn't get last transaction"))?,
+        )?
+    }
+    if tailing {
+        // listening for incoming transactions
+        loop {
+            thread::sleep(time::Duration::from_millis(1000));
+            let resp =
+                get_account_transactions_response(&client, address, &network, old_seq_num + 1, 100)
+                    .await?;
+            let json_with_txns: serde_json::Value =
+                serde_json::from_str(resp.text().await?.as_str())?;
+            let txn_array = json_with_txns
+                .as_array()
+                .ok_or_else(|| anyhow!("Couldn't convert to array"))?
+                .to_vec();
+
+            // checking if there are transactions
+            if txn_array.is_empty() {
+                continue;
+            }
+            let last_txn_seq_num = parse_txn_for_seq_num(
+                txn_array
+                    .last()
+                    .ok_or_else(|| anyhow!("Couldn't get last transaction"))?,
+            )?;
+            write_out_txns(txn_array, last_txn_seq_num, old_seq_num, &mut io::stdout())?;
+            old_seq_num = last_txn_seq_num;
+        }
+    }
+    Ok(())
+}
+
+fn write_out_txns<W: Write>(
+    all_transactions: Vec<Value>,
+    last_txn_seq_num: i64,
+    old_seq_num: i64,
+    mut stdout: W,
+) -> Result<()> {
+    if last_txn_seq_num > old_seq_num {
+        for txn in all_transactions.iter() {
```

Comment:
Building off the previous comment and the notion of single responsibility, why have a seq_num check inside here if it's job is to write_out_txns?

```rust
fn write_out_txns<W: Write>(txns, mut stdout) {} // is clean
```